frontend/src/app/chart/chart.component.spec.ts

```diff
@@ -13,26 +13,136 @@
 // =============================================================================
 
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { from, empty } from 'rxjs';
 
 import { ChartComponent } from './chart.component';
+import { HttpService, STRATEGY } from '../services/http.service';
+import { Record } from '../record';
+
+const getMaxMin = (records: number[]) => {
+  return [
+    records.reduce((left, right) => (left < right ? left : right)),
+    records.reduce((left, right) => (left > right ? left : right)),
+  ];
+};
+
+const generateNewData = (length: number) => {
+  const newData: [number, number, string][] = [];
+  for (let index = 0; index < 600; index++) {
+    newData.push([1573149236356988 + index * 2000, Math.random() * 500, 'sys']);
```

Comment:
This line of code looks confusing? any reasons to generate random data with the specific number?

Reply:
Since we are working on Unix timestamp, the timestamp must be equal or later than the unix epoch, and `1573149236356988` is just a random number I choose from the csv file to make it reasonable.
And yes in `index * 2000`, it could be more randomized. But I think for the purpose of testing the chart domains, it should be enough.